spec/requests/users_controller_spec.rb

```diff
@@ -4080,6 +4080,43 @@ def create_totp
     end
   end
 
+  describe '#disable_second_factor' do
+    context 'when logged in with secure session' do
+      before do
+        sign_in(user)
+        stub_secure_session_confirmed
+      end
+      context 'when user has a registered totp and security key' do
+        before do
+          expect(user.reload.user_second_factors).to be_empty
+          expect(user.reload.security_keys).to be_empty
+
+          totp_second_factor = Fabricate(:user_second_factor_totp, user: user)
+          security_key_second_factor = Fabricate(:user_security_key, user: user, factor_type: UserSecurityKey.factor_types[:second_factor])
+
+          expect(user.reload.user_second_factors.totps.first).to eq(totp_second_factor)
+          expect(user.reload.security_keys.first).to eq(security_key_second_factor)
+        end
+
+        it 'should disable all totp and security keys' do
+          expect do
+            put "/u/disable_second_factor.json"
+          end.to change { Jobs::CriticalUserEmail.jobs.length }.by(1)
+
+          expect(response.status).to eq(200)
+
+          expect(user.reload.user_second_factors).to be_empty
+          expect(user.reload.security_keys).to be_empty
+
+          job_args = Jobs::CriticalUserEmail.jobs.first["args"].first
```

Comment:
Let's use the helpers in https://github.com/discourse/discourse/blob/0967ce478d85ad5053522944b36ed9212f867457/spec/support/sidekiq_helpers.rb#L19